File: examples/sample_code.c

#include <stdio.h>
#include <stdlib.h>

// 计算阶乘的函数
int factorial(int n) {
    if (n <= 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

// 主函数
int main() {
    int num = 5;
    int result;
    
    printf("计算 %d 的阶乘\n", num);
    result = factorial(num);
    printf("%d! = %d\n", num, result);
    
    return 0;
}

File: examples/sample_code.pas

program FactorialExample;

{ 计算阶乘的函数 }
function Factorial(n: integer): integer;
begin
    if n <= 1 then
        Factorial := 1
    else
        Factorial := n * Factorial(n - 1);
end;

{ 主程序 }
var
    num, result: integer;
begin
    num := 5;
    writeln('计算 ', num, ' 的阶乘');
    result := Factorial(num);
    writeln(num, '! = ', result);
end.
